example_plugins/shift_notes_right_x_lanes/plugin.lua

```diff
@@ -0,0 +1,61 @@
+function draw()
+    imgui.Begin("Move Selected Notes Right x Lanes")
+
+    --Prevent the user from interacting with the editor when interacting with the plugin window
+    state.IsWindowHovered = imgui.IsWindowHovered()
+
+    local lanes = 0
+
+    --Determine the number of lanes in the map
+    if map.Mode == game_mode.Keys4 then
+        lanes = 4
+    elseif map.Mode == game_mode.Keys7 then
+        lanes = 7
+    else
+        imgui.Text("Invalid keymode!")
+        imgui.End()
+        return
+    end
+
+    --Retrieve the value of x, defaulting to 1
+    local x = state.GetValue("x") or 1
+
+    local selection = state.SelectedHitObjects
+
+    --The first returned value is irrelevant
+    --The second returned value is what the user inputs
+    _, x = imgui.InputInt("", x)
+
+    --[[
+        Buttons convey their "pressed" state with a
+        boolean value, which can be directly used in an
+        if-condition.
+    ]]
+
+    if imgui.Button("Shift Notes " .. x .. plural(" Lane", x) .." to the Right") then
+        local result = {}
+
+        for _, note in pairs(selection) do
```

Comment:
Since the example plugins are supposed to be a tutorial/guide or similar, replacing the underscore with i to clarify that it's the key/index of the for loop might help with understanding the for-loop with pairs().
```suggestion
        for i, note in pairs(selection) do
```